File: DSA/pranavdixit_590014418/Day_12_Question_3.py
def compress_string(s):
    if not s:
        return s
    result = ''
    count = 1
    for i in range(1, len(s)):
        if s[i] == s[i - 1]:
            count += 1
        else:
            result += s[i - 1] + str(count)
            count = 1
    result += s[-1] + str(count)
    return result if len(result) < len(s) else s

print(compress_string("aabcccccaaa"))
print(compress_string("abc"))
print(compress_string("aabbcc"))

File: DSA/AYUSHMEHTA_590015131/Day12_Question3.cpp
#include <iostream>
#include <string>
using namespace std;

string compressString(const string &s) {
    string compressed = "";
    int count = 1;

    for (int i = 1; i < s.length(); i++) {
        if (s[i] == s[i - 1]) {
            count++;
        } else {
            compressed += string(1, s[i - 1]) + to_string(count);
            count = 1;
        }
    }

    compressed += string(1, s.back()) + to_string(count);

    return compressed.length() < s.length() ? compressed : s;
}

int main() {
    string s;
    cout << "Enter a string: ";
    getline(cin, s);

    string result = compressString(s);
    cout << "Compressed: " << result << endl;

    return 0;
}
